### PR TITLE
Improve command-line interface.

### PR DESCRIPTION
**Pull request by nashef**
This converts the rosette cmd-line interface to using a more standard gnu-like option set, adds a `--help`, and cleans up the usage message quite a bit.

### CODE REVIEW

**Comment by kjmcqueen** [comment]
Previously, excess arguments were passed on to the RBL code.
Does this support that?
Is there no longer a way to do that?
Is it not necessary to support that?

**Comment by nashef** [comment]
> PLS update run.sh

I didn't modify run.sh. Nor do I use it.

> Previously, excess arguments were passed on to the RBL code.

I left a comment explaining that I wasn't sure what to do, but I believe that this is functionally the same as the old logic. It returns an `argc` value that points at the next option after the last processed one.

I suspect the real answer is somewhat more subtle, but I am unsure. Do we have a test or some code that uses this feature? I can run it and see how it works before/after.

**Comment by kjmcqueen** [comment]
The rbl files loaded by boot.rbl do make use of the arguments, and there are a couple of arguments looked for, but they are accessed by value, not position. Beyond that, I'm not sure how they are used.

-ess_sysdir <path>
-docs

Attempting to use either of these causes your new parser to spit out an error and usage message.

**Comment by nashef** [comment]
I believe that I fixed the problem, @kjmcqueen. It should work like this, now:

`rosette --boot=boot.rbl my_program.rbl these --are=program args`
`rosette --boot=boot.rbl -- these are program args`

It doesn't seem like the `RunFile` is really working, but I am not sure that's actually a change. Did it work before?

File: rosette/src/CommandLine.cc [hunk 1]
```diff
@@ -71,114 +97,117 @@ int ParseCommandLine(int argc, char** argv) {
...
-                usage(cmd_name);
-            }
-
-            DeleteArgs(i, argc, argv, 2);
-            argc -= 2;
-            continue;
-        }
+    while (1) {
+        int option_index = 0;
+        c = getopt_long(argc, argv, "+qt:p:i:s:o:b:",
```
Comment (kirkwood) [inline rosette/src/CommandLine.cc]: Little `i` takes no argument; big `I` missing, `h` missing, `v` missing, `d` missing.

**Comment by kirkwood** [comment]
I am loving this change for sure.

File: rosette/src/CommandLine.cc (hunk 1 above)
Comment (nashef) [reply rosette/src/CommandLine.cc]: Thank you! Fix incoming...

**Comment by kjmcqueen** [comment]
Previously, the REPL was enabled even if a RunFile was specified. It would just drop into the REPL after loading the RunFile. If you didn't want a REPL, you could add a (exit) at the end of your rbl file. I like the idea of it being controlled from the rosette command line. I don't have a preference for whether it defaults to ON or OFF.

File: rosette/src/CommandLine.cc (hunk 1 above)
Comment (nashef) [reply rosette/src/CommandLine.cc]: Done.

File: rosette/src/CommandLine.cc [hunk 2]
```diff
@@ -71,114 +97,117 @@ int ParseCommandLine(int argc, char** argv) {
...
-    const char* cmd_name = argv[0];
+    int verbose_flag = 0;
+    int enable_repl = 0;
+    int c = 0;
+
+    const struct option long_options[] = {
+        /* Flags */
+        {"verbose", no_argument, &verbose_flag, 1},
+        {"quiet", no_argument, &verbose_flag, 0},
+        {"interactive-repl", no_argument, &enable_repl, 'i'},
```
[the remark this answers is not in the document]
Comment (nashef) [reply rosette/src/CommandLine.cc]: Fixed.

**Comment by nashef** [comment]
@kirkwood Okay, I think I fixed the argument handling, but you're right it doesn't fix everything it could. In the interest of keeping each PR small, I'd like to merge and then do the rest separately. It doesn't seem too hard, but it'll require some work.

**Pull request merged**